libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/TxBodySpec.hs

```diff
@@ -0,0 +1,322 @@
+  putPretty "\nSubMap of Utxo" (UTxO subutxo)
+  putStrLn ("SubMap of UTxO, total Coin " ++ show (coinBalance @era (UTxO subutxo)))
+  putPretty "\nfeeInput" feeinput
+  putPretty "\nTxBody" basic
+
+sumCoin_ :: forall fn era. EraSpecTxOut era fn => Term fn [TxOut era] -> Term fn Coin
+sumCoin_ x = foldMap_ (txOutCoin_ @era @fn) x
+
+-- ===============================================================
+
+bodyspec2 ::
+  forall era fn.
+  ( EraSpecTxOut era fn
+  , EraSpecCert era fn
+  ) =>
+  CertsEnv era ->
+  CertState era ->
+  Specification
+    fn
+    ( ShelleyTxBody era
+    , Map (TxIn (EraCrypto era)) (TxOut era)
+    , TxIn (EraCrypto era)
+    )
+bodyspec2 certsenv certstate =
+  constrained' $ \ [var|shelleyBody|] [var|utxo|] [var|feeInput|] ->
+    match shelleyBody $
+      \ [var|inputs|] [var|outputs|] [var|certs|] [var|rewAcct|] [var|fee|] _ [var|update|] _ ->
+        [ dependsOn shelleyBody fee
+        , dependsOn shelleyBody (inputs :: Term fn (Set (TxIn (EraCrypto era))))
+        , dependsOn shelleyBody outputs
+        , dependsOn shelleyBody certs
+        , dependsOn utxo inputs
+        , -- , exists (\eval -> undefined) $ \ [var|feeInput|] ->
+          exists (\eval -> pure (Map.restrictKeys (eval utxo) (eval inputs))) $ \ [var|utxosubset|] ->
+            exists (\_eval -> undefined) $ \ [var|tempUtxo|] ->
+              [ dependsOn inputs utxosubset
+              , dependsOn utxo utxosubset
+              , dependsOn feeInput inputs
+              , dependsOn fee utxosubset
+              , dependsOn fee feeInput
+              , dependsOn outputs utxosubset
+              , dependsOn outputs fee
+              , dependsOn utxo tempUtxo
+              , satisfies utxosubset (hasSize (rangeSize 3 4))
+              , assert $ member_ feeInput inputs
+              , assert $ inputs ==. dom_ utxosubset
+              , assert $ onJust (lookup_ feeInput utxosubset) (\ [var|txout|] -> fee ==. txOutCoin_ @era @fn txout)
+              , satisfies (dom_ tempUtxo) (hasSize (rangeSize 8 10))
+              , subMapSuperDependsOnSub utxosubset tempUtxo
+              , assert $ (sumCoin_ @fn @era outputs) ==. sumCoin_ @fn @era (rng_ utxosubset) - fee
+              , forAll outputs $ \x -> txOutCoin_ @era @fn x >=. lit (Coin 0)
+              , reify
+                  (pair_ tempUtxo feeInput)
+                  (\(m, i) -> Map.adjust baz i m)
+                  (\u -> utxo ==. u)
+              ]
+        , assert $ update ==. lit Nothing
+        , forAll certs $ \ [var|oneCert|] -> satisfies oneCert (txCertSpec (projectEnv certsenv) certstate)
+        , assert $ sizeOf_ certs <=. 4
+        , assert $ rewAcct ==. lit Map.empty
+        , assert $ sizeOf_ outputs ==. 4
+        ]
+
+baz :: EraTxOut era => TxOut era -> TxOut era
+baz x = x & coinTxOutL .~ ((x ^. coinTxOutL) <+> (Coin 100))
+
+go2 ::
+  forall era.
+  ( EraSpecTxOut era ConwayFn
+  , EraSpecCert era ConwayFn
+  , HasSpec ConwayFn (Tx era)
+  ) =>
+  IO ()
```

Comment:
Same issue as above with the names of tests

Reply:
so added to LedgerTypes.Tests with a better name. An all those tests are run Nightly